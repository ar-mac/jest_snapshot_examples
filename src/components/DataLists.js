import React from "react";
import PropTypes from "prop-types";
import cn from "classnames";

const DataLists = ({users, posts}) => {
  const wrapperClass = cn({
    "both_lists": users.length && posts.length,
    "single_or_no_lists": !(users.length && posts.length),
  });
  const classDependentOnUsers = cn({
    no_users: users.length === 0,
    single_user: users.length === 1,
    some_users: users.length > 1 && users.length <= 3,
    lots_of_users: users.length > 5,
  });

  return (
    <div className={wrapperClass}>
      {
        users.length ? (
          <ul>{users.map((user, index) => (
            <li key={index}>
              <span>User: </span>
              <span>{user}</span>
            </li>
          ))}</ul>
        ) : (
          <div>No users to display</div>
        )
      }
      {
        posts.length ? (
          <ul>{posts.map((post, index) => (
            <li key={index}>
              <span>Post: </span>
              <span>{post}</span>
            </li>
          ))}</ul>
        ) : (
          <div className={classDependentOnUsers}>No posts to display</div>
        )
      }
    </div>
  )
};

DataLists.propTypes = {
  users: PropTypes.array,
  posts: PropTypes.array,
};

DataLists.defaultProps = {
  users: [],
  posts: [],
};

export default DataLists;
