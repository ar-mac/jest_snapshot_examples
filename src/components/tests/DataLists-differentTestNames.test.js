import React from "react";
import { shallow } from "enzyme";
import DataLists from "../DataLists";

describe("DataLists component", () => {
  // BAD - generic and not helpful test name
  it("renders data list correctly", () => {
    const users = ["user1"];
    expect(shallow(<DataLists users={users} />)).toMatchSnapshot();
  });

  // LITTLE BETTER - test name at least suggests what is the reason of the test
  it("renders single user without posts and with single_user class", () => {
    const users = ["user1"];
    expect(shallow(<DataLists users={users} />)).toMatchSnapshot();
  });
});
