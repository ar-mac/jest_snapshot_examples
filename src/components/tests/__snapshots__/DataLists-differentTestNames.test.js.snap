// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLists component renders data list correctly 1`] = `
<div
  className="single_or_no_lists"
>
  <ul>
    <li
      key="0"
    >
      <span>
        User: 
      </span>
      <span>
        user1
      </span>
    </li>
  </ul>
  <div
    className="single_user"
  >
    No posts to display
  </div>
</div>
`;

exports[`DataLists component renders single user without posts and with single_user class 1`] = `
<div
  className="single_or_no_lists"
>
  <ul>
    <li
      key="0"
    >
      <span>
        User: 
      </span>
      <span>
        user1
      </span>
    </li>
  </ul>
  <div
    className="single_user"
  >
    No posts to display
  </div>
</div>
`;
